scripts/west_commands/ncs_west_helpers.py

```diff
@@ -235,84 +262,86 @@ def _downstream_outstanding_commits(self):
                     level=log.VERBOSE_VERY)
 
         return list(downstream_out.values())
 
-    def _likely_merged_commits(self):
+    def _likely_merged_commits(self) -> dict[pygit2.Commit,
+                                             list[pygit2.Commit]]:
         # Compute patches which are downstream and probably were
         # merged upstream, using the following heuristics:
         #
-        # 1. downstream patches with small shortlog edit distances
+        # 1. downstream patches with small title edit distances
         #    from upstream patches
         #
-        # 2. downstream patches with shortlogs that are prefixes of
+        # 2. downstream patches with titles that are prefixes of
         #    upstream patches
         #
-        # Heuristic #1 catches patches with typos in the shortlogs
+        # Heuristic #1 catches patches with typos in the titles
         # that reviewers asked to be fixed, etc. E.g. upstream
-        # shortlog
+        # title
         #
         #    Bluetoth: do foo
         #
-        # matches downstream shortlog
+        # matches downstream title
         #
         #    [nrf fromlist] Bluetooth: do foo
         #
         # Heuristic #2 catches situations where we had to shorten our
-        # downstream shortlog to fit the "[nrf xyz]" sauce tag at the
-        # beginning and still fit within CI's shortlog length
-        # restrictions. E.g. upstream shortlog
+        # downstream title to fit the "[nrf xyz]" sauce tag at the
+        # beginning and still fit within CI's title length
+        # restrictions. E.g. upstream title
         #
         #    subsys: do a thing that is very useful for everyone
         #
-        # matches downstream shortlog
+        # matches downstream title
         #
         #    [nrf fromlist] subsys: do a thing that is very
         #
         # The return value is a map from pygit2 commit objects for
         # downstream patches, to a list of pygit2 commit objects that
-        # are upstream patches which have similar shortlogs and the
+        # are upstream patches which have similar titles and the
         # same authors.
 
-        likely_merged = OrderedDict()
+        likely_merged = {}
         for dc in self.downstream_outstanding:
-            sl = commit_shortlog(dc)
+            sl = commit_title(dc)
 
             def ed(upstream_commit):
                 return editdistance.eval(
-                    shortlog_no_sauce(sl, self._downstream_sauce),
-                    commit_shortlog(upstream_commit))
+                    title_no_sauce(sl, self._downstream_sauce),
+                    commit_title(upstream_commit))
 
             matches = [
                 uc for uc in self.upstream_new if
                 # Heuristic #1:
                 ed(uc) < self._edit_dist_threshold or
                 # Heuristic #2:
-                commit_shortlog(uc).startswith(sl)
+                commit_title(uc).startswith(sl)
```

Comment:
I had PR somewhere fixing this but I guess it was not merged. So now that you are touching this line heuristic 2 could be fixed:
commit_title(uc).startswith(sl) --> commit_title(uc).startswith(title_no_sauce(sl, self._downstream_sauce))

Reply:
> So now that you are touching this line heuristic 2 could be fixed:

@karhama I think this check should be eliminated entirely; see `scripts: west_commands: fix and optimize ncs-compare heuristic`